udf/add_searches_by_index.sql

```diff
@@ -0,0 +1,80 @@
+/*
+
+Return sums of each search type grouped by the index.  Results are ordered by index.
+
+ */
+CREATE OR REPLACE FUNCTION udf.add_searches_by_index(
+  searches ARRAY<
+    STRUCT<
+      total_searches INT64,
+      tagged_searches INT64,
+      search_with_ads INT64,
+      ad_click INT64,
+      index INT64
+    >
+  >
+) AS (
+  (
+    WITH summed_by_index AS (
+      SELECT AS STRUCT
+        SUM(search.total_searches) AS total_searches,
+        SUM(search.tagged_searches) AS tagged_searches,
+        SUM(search.search_with_ads) AS search_with_ads,
+        SUM(search.ad_click) AS ad_click,
+      FROM
+        UNNEST(searches) search
+      GROUP BY
+        index
+      ORDER BY
+        index
+    )
+    SELECT AS STRUCT
+      ARRAY_AGG(total_searches) total_searches,
```

Comment:
The ARRAY_AGG docs state:

> If ORDER BY is not specified, the order of the elements in the output array is non-deterministic, which means you might receive a different result each time you use this function

So I think we need to put the ORDER BY in the ARRAY_AGG calls rather than in the CTE above to guarantee ordering (even though your tests seem to indicate ordering is stable).